packages/components/src/components/ColorPicker/ColorPicker.tsx

```diff
@@ -0,0 +1,100 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { Slot as RadixSlot } from '@radix-ui/react-slot';
+import { Children, forwardRef, useCallback, useState, type ReactNode } from 'react';
+
+import { ForwardedRefColorGradientInput } from './ColorGradientInput';
+import { ForwardedRefColorPickerInput } from './ColorPickerInput';
+import { ForwardedRefColorValueInput } from './ColorValueInput';
+import styles from './styles/customColorPicker.module.scss';
+import { type ColorFormat, type RgbaColor } from './types';
+import { DEFAULT_COLOR, DEFAULT_FORMAT, rgbColorToHex } from './utils';
+
+type ColorPickerProps = {
+    /**
+     * The children of the color picker component. This can contain multiple `ColorPicker.Values` or `ColorPicker.Gradient` components.
+     */
+    children?: ReactNode;
+    /**
+     * The active color in the color picker
+     */
+    currentColor?: RgbaColor;
+    /**
+     * Event handler called when the color changes
+     */
+    onColorChange?: (color: RgbaColor) => void;
+    /**
+     * The default format to use for the color input
+     * @default "HEX"
+     */
+    defaultFormat?: ColorFormat;
+};
+
+export const ColorPickerRoot = ({
+    children,
+    currentColor = DEFAULT_COLOR,
+    onColorChange = () => {},
+    defaultFormat = DEFAULT_FORMAT,
+    ...props
+}: ColorPickerProps) => {
+    const [currentFormat, setCurrentFormat] = useState<ColorFormat>(defaultFormat);
+
+    const handleColorChange = useCallback(
```

Comment:
if this function received a `onColorChange` callback as well you could pull it out of this closure entirely and avoid having to memoize it. Since you have to receive `currentFormat` anyway it might not be a bad idea to add that 3rd param